### ISSUE TITLE
Improved handling of file-extension conflicts

### ISSUE THREAD

**Issue by sharkdp** [sharkdp/bat#1076]
With #1070 merged, we now have four known exceptions to the "no duplicate extensions" rule (see `tests/no_duplicate_extensions.rs`). We have a separate test (`syntax_detection_well_defined_mapping_for_duplicate_extensions`) that makes sure that these extensions are handled as expected (mapped to the desired syntax). For example:
``` rust
assert_eq!(test.syntax_for_file("test.sass"), "Sass");
```

However, the duplicate file extensions still appear in the `bat --list-languages` output for both languages:
```
…
Ruby Haml                            haml, sass
…
Sass                                 sass
…
```
In order to prevent this, when printing the list of syntaxes, we could go through each pattern that is displayed and run it through our syntax mapping function to make sure that it's actually mapped to the current syntax. When printing the extensions for `Ruby Haml`, we would check the `sass` pattern (by getting the syntax for a file named `test.sass` AND for a file named `sass`) and find the `Sass` syntax instead. We could then simply skip that pattern.

**Comment by sharkdp** [comment]
Fixed by @Kienyew in #1135.

**Comment by sharkdp** [comment]
Fixed in [v0.16.0](https://github.com/sharkdp/bat/releases/tag/v0.16.0).
